src/Filesystem/Provider/LocalFilesystemFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\Filesystem;
+use ILIAS\Filesystem\Provider\Configuration\LocalConfig;
+
+/**
+ * Class LocalFilesystemFactory
+ *
+ * The local filesystem factory creates instances of the local filesystem adapter.
+ * The purpose of the specific factory is to hold the FilesystemFactories clean from specific bootstrap code of each adapter.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ *
+ * @see FilesystemFactory
+ */
+interface LocalFilesystemFactory {
```

Comment:
Yes thats the interface of the local filesystem factory. But you are absolutely right it has nothing todo with the public interface sorry.